server/leafnode.go

```diff
@@ -628,7 +628,12 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 
 		c.mu.Unlock()
 		// Error will be handled below, so ignore here.
-		c.parse([]byte(info))
+		err = c.parse([]byte(info))
+		if err != nil {
+			c.Debugf("Error reading remote leafnode's INFO: %s", err)
```

Comment:
Agree, I was thinking about that as right now unless using debug it may look as if it was connected. I used debug to be similar to how this error was handled: https://github.com/nats-io/nats-server/pull/1265/files/ecb5008fe33b5373e95183a6fc5fb68345a4a0eb#diff-51766556c6c02cb21c6ac4f9c3fbe375R641

Reply:
Should be changed to an error too, IMO.